src/App.tsx

```diff
@@ -1,73 +1,133 @@
-import React from 'react';
+/* eslint-disable @typescript-eslint/indent */
+import React, { useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import classNames from 'classnames';
+import { Person } from './types/Person';
+import { useEffect, useRef } from 'react';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [inputFocused, setInputFocused] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [debounceDelay, setDebounceDelay] = useState(300); // Customizable delay
+  const dropdownRef = useRef<HTMLDivElement | null>(null);
+
+  const useDebounce = <T,>(value: T, delay = 300): T => {
+    const [debouncedValue, setDebouncedValue] = useState(value);
+
+    useEffect(() => {
+      const handler = setTimeout(() => setDebouncedValue(value), delay);
+
+      return () => clearTimeout(handler);
+    }, [value, delay]);
+
+    return debouncedValue;
+  };
+
+  const debouncedQuery = useDebounce(searchQuery, debounceDelay);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchQuery(event.target.value);
+    setSelectedPerson(null);
+  };
+
+  const handlePersonSelect = (newTarget: Person) => {
+    setSelectedPerson(newTarget);
+    setInputFocused(false); // Close the dropdown after selecting a person
+  };
+
+  // Handle click outside
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        dropdownRef.current &&
+        !dropdownRef.current.contains(event.target as Node)
+      ) {
+        setInputFocused(false); // Close the dropdown
+      }
+    };
+
+    document.addEventListener('mousedown', handleClickOutside);
+
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, []);
+
+  // Filter people based on the debounced query
+  const filteredPeople = peopleFromServer.filter(person =>
+    debouncedQuery
+      ? person.name.toLowerCase().includes(debouncedQuery.toLowerCase())
+      : inputFocused,
```

Comment:
The condition for filtering `peopleFromServer` is incorrect. The `inputFocused` state should not be used as a fallback condition for filtering. This can lead to unexpected behavior when the input is focused but the query is empty. Consider using only the `debouncedQuery` for filtering.